docs/DeveloperGuide.md

```diff
@@ -283,32 +283,104 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TutorAid` and the **Actor** is the `tutor`, unless specified otherwise)
 
-**Use case: Delete a person**
+**Use case 5: Updating payment for student**
+
+Preconditions: There is at least one student added to TutorAid.
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User views list of students
+2.  User identifies the student index to update payment record
+3.  User updates payment for student
+4.  TutorAid successfully updates the payment for a student corresponding to the given index number.
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
+* 1a. Invalid list command syntax
+  
+    * 1a1. TutorAid displays an error message and requests the tutor to re-enter the command.
+    * 1a2. User re-enters the command.
 
-  Use case ends.
+      Steps 1a1-1a2 are repeated until the syntax entered is correct.
+      Use case resumes from step 2.
 
-* 3a. The given index is invalid.
+* 3a. Invalid paid/unpaid command syntax
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. TutorAid displays an error message and requests the tutor to re-enter the command.
+    * 3a2. User re-enters the command
 
-      Use case resumes at step 2.
+      Steps 3a1 - 3a2 are repeated until a valid syntax is given
+      Use case resumes from step 4
 
-*{More to be added}*
+* 3b. The given student index is invalid
+ 
+    * 3b1. TutorAid shows an error message
+    * 3b2. User enters a new index with paid/unpaid command
+  
+      Steps 3a1 - 3a2 are repeated until a valid index is given
+      Use case resumes from step 4
+    
+* 4a. Incorrect update of payment for a different student instead
```

Comment:
I was wondering if extension 4a is needed? This is because extensions usually describe exceptional/alternative flow of events, where there is a disruption when going through the steps stated in the MSS. But for 4a the tutor will technically be able to complete the whole MSS although the payment update is being made for the wrong student?

Otherwise LGTM